src/pint/toa.py

```diff
@@ -1491,31 +1532,41 @@ def get_dm_errors(self):
             raise AttributeError("No DM error is provided.")
         return np.array(result)[valid] * pint.dmu
 
-    def get_groups(self, gap_limit=None):
+    def get_groups(self, gap_limit=2 * u.h, add_column=False):
         """Flag toas within gap limit (default 2h = 0.0833d) of each other as the same group.
 
         Groups can be larger than the gap limit - if toas are separated by a gap larger than
         the gap limit, a new group starts and continues until another such gap is found.
 
-        Groups with a two-hour spacing are pre-computed when the TOAs object is constructed,
+        Groups with a two-hour spacing (default) are pre-computed when the TOAs object is constructed,
```

Comment:
I think this is no longer true with this change? Perhaps an updated summary of the caching functionality would be more appropriate? (If a matching column exists it will be used; `add_column` allows construction of such a column.)